localstack-core/localstack/utils/container_utils/docker_cmd_client.py

```diff
@@ -266,17 +266,21 @@ def commit(
             ) from e
 
     def remove_container(self, container_name: str, force=True, check_existence=False) -> None:
-        if check_existence and container_name not in self.get_running_container_names():
+        if check_existence and container_name not in self.get_all_container_names():
             return
         cmd = self._docker_cmd() + ["rm"]
         if force:
             cmd.append("-f")
         cmd.append(container_name)
         LOG.debug("Removing container with cmd %s", cmd)
         try:
-            run(cmd)
+            output = run(cmd)
+            # When the container does not exist, the output could have the error message without any exception
+            if isinstance(output, str) and not force:
```

Comment:
Does the output have the error message if force is set? I am fine with checking this, just wondering